The Grouse Outdoors

AndrewL97 opened the pull request:
This PR adds outdoor effects to MS in the form of Weather, using particle effects, and Sunlight, which adds to the lighting system.

Each turf is given an Outdoor_effects object, that holds up to two overlays - one for weather and one for sunlight.

Sunlight is governed by three "states" on a turf, based on a roof_type 
SKY_BLOCKED
SKY_VISIBLE
SKY_VISIBLE_BORDER

Rather than setting colours on individual overlays for each turf - we use a greyscale overlay, and a coloured fullscreen object to colour everything. This means that we can animate sunlight changes, without completely melting the server.

Weather is affected by whether or not the roof above is weatherproof - for now this is limited to space/open_space

AndrewL97 committed:
The great outdoors (#6)

AndrewL97 committed:
Fix eslint panic

AndrewL97 committed:
no more

AndrewL97 committed:
Merge branch 'master' of https://github.com/Mojave-Sun/mojave-sun-13 into TheGrouseOutdoors

AndrewL97 commented:
Lint issues regarding particle effects are lies 
fadein is a native byond var for /particles
and generators can be chained together

Koshenko commented on mojave/modules/outdoor_effects/code/modules/particool/particool.dm:
```diff
@@ -0,0 +1,203 @@
+/datum/particle_editor/proc/stringToList(str, toNum = FALSE)
+	. = splittext(str,regex(@"(?<!\\),"))
+	if(toNum)
+		for(var/i = 1; i <= length(.); ++i)
```
Could probably optimize this slightly with `for(i in list)

Koshenko commented on mojave/modules/outdoor_effects/code/_onclick/hud/fullscreen.dm:
```diff
@@ -0,0 +1,21 @@
+/* Our sunlight planemaster mashes all of our sunlight overlays together into one             */
+/* The fullscreen then grabs the plane_master with a layer filter, and colours it             */
+/* We do this so the sunlight fullscreen acts as a big lighting object, in our lighting plane */
+/atom/movable/screen/fullscreen/lighting_backdrop/Sunlight
```
Lowercase

Koshenko commented on mojave/modules/outdoor_effects/code/controllers/subsystem/outdoor_effects.dm:
```diff
@@ -0,0 +1,307 @@
+//  color = "#050d29"
+//  start = "0" //12:00:00 AM
+
+/datum/time_of_day/Dawn
```
Lowercase all the datums in here

Koshenko commented on mojave/modules/outdoor_effects/code/controllers/subsystem/particle_weather.dm:
```diff
@@ -0,0 +1,84 @@
+	weatherEffect.particles = particleEffect
+
+/datum/controller/subsystem/ParticleWeather/proc/stopWeather()
+	qdel(runningWeather)
```
`QDELL_NULL()` is a nice shorthand macro for `qdel` then set the reference to null.

Koshenko reviewed (changes requested):
Added a few small requests for changes.

AndrewL97 committed:
change weatherproof from glob list to turf var
fix particle icon
fix some class name capitalization
uses qdel_null

todo - investigate hard deletes

AndrewL97 committed:
I think I have done it

Koshenko committed:
Merge branch 'master' into TheGrouseOutdoors

Koshenko committed:
Removes usage of deprecated radiation code
